### PR TITLE
feat: add option to customize Go migration template

### PR DESCRIPTION
**Pull request by greeflas**
Discussion #795

### CODE REVIEW

File: migrate/migrator.go [hunk 1]
```diff
@@ -227,12 +228,19 @@ func WithPackageName(name string) GoMigrationOption {
 	}
 }
 
+func WithContent(content string) GoMigrationOption {
```
Comment (vmihailenco) [inline migrate/migrator.go:231]: Please rename to `WithGoTemplate` so we can also have `WithSQLTemplate` later.

File: migrate/migrator.go [hunk 2]
```diff
@@ -217,6 +217,7 @@ func (m *Migrator) Rollback(ctx context.Context, opts ...MigrationOption) (*Migr
 
 type goMigrationConfig struct {
 	packageName string
+	content     string
```
Comment (vmihailenco) [inline migrate/migrator.go:220]: content -> goTemplate

File: migrate/migrator.go (hunk 1 above)
Comment (greeflas) [reply migrate/migrator.go]: Done.

File: migrate/migrator.go (hunk 2 above)
Comment (greeflas) [reply migrate/migrator.go]: Done.

**Comment by vmihailenco** [comment]
Thanks :+1: Ideally, we should also start using text/template here to make code generation more flexible. Guess this can done in a separate PR later - https://github.com/uptrace/bun/issues/797

**Pull request merged**